### ISSUE TITLE
Lots of unnecessarily required config fields

### ISSUE THREAD

**Issue by werdnum** [davidusb-geek/emhass#262]
**Describe the bug**
When I tried to start up EMHASS for the first time, I deleted all the parts of the configuration file that didn't apply (I have no solar and no battery). EMHASS refused to start up and/or do any optimisation until I added them all back.

This makes config migrations harder when you add new required options (as also happened later on)

**To Reproduce**
Run EMHASS with a configuration file that includes no parameters related to solar and batteries and try to do a forecast. 

**Expected behavior**

EMHASS should ignore the absence of fields that are not relevant to the current configuration.

If EMHASS has required fields, it should throw a user friendly error on startup instead of KeyError on doing a forecast.

**Screenshots**
N/A

**Home Assistant installation type**
Container, but not relevant.

**Your hardware**
Linux Kubernetes arm64

**EMHASS installation type**
Container in Kubernetes

**Additional context**
Maybe worth using whatever HA uses for configuration validation?

**Comment by davidusb-geek** [comment]
When was this first time that you tried? There has been a lot of improvements for these types of issues thanks to the work of @GeoDerp earlier this year. There is now a treatment for setting default values when they are not specified by the user.
The configuration file in the add-on was also updated and rearranged for better treatment of optional and non-optional params. But there is probably more room for improvement. Although this is foe the add-on installation method and not the docker standalone, so that may not be completely irrelevant

**Comment by GeoDerp** [comment]
> When was this first time that you tried? There has been a lot of improvements for these types of issues thanks to the work of @GeoDerp earlier this year. There is now a treatment for setting default values when they are not specified by the user.
> The configuration file in the add-on was also updated and rearranged for better treatment of optional and non-optional params. But there is probably more room for improvement. Although this is foe the add-on installation method and not the docker standalone, so that may not be completely irrelevant 

Hey @werdnum, you probably have some good ideas, but as @davidusb-geek already mentioned we have done some parameter defaulting in the add-on version of EMHASS. *Standalone so far we just assumed people left the defaults in.* (not saying we can't change this)
If you like, try running the add-on version of EMHASS and see what you think. You will need to convert your parameters *(including their respective parameters name changes)* from the config.yaml to the options.yaml. 
I'll update this comment Tomorrow with more details on this. But for now have a look at https://emhass.readthedocs.io/en/latest/develop.html for docker run command difference, and https://emhass.readthedocs.io/en/latest/differences.html for the parameter name differences. 
Hope that helped 😁

**Comment by GeoDerp** [comment]
If you with to grab the image via a container registry: *(not build the image locally)*
- EMHASS amd64 add-on: https://hub.docker.com/r/davidusb/image-aarch64-emhass
- Docker run example: 
  ```bash
  docker run -it -p 5000:5000 --name emhass-container -v $(pwd)/options.json:/app/options.json  davidusb/image-aarch64-emhass:latest 
  ```
  *This assumes secrets are passed via options.json. Alternately, you can mount the secrets file also and EMHASS should reference it.*
  ```bash
  docker run -it -p 5000:5000 --name emhass-container -v $(pwd)/options.json:/app/options.json  -v $(pwd)/secrets_emhass.yaml:/app/secrets_emhass.yaml davidusb/image-aarch64-emhass:latest 
  ```

**Comment by werdnum** [comment]
I'm not sure I understand the difference between the 'addon' and the regular version of EMHASS. I thought addons were just for HAOS, and I'm not using HAOS, so I am just using the normal version. Is the addon version better supported or something?

**Comment by GeoDerp** [comment]
> I'm not sure I understand the difference between the 'addon' and the regular version of EMHASS. I thought addons were just for HAOS, and I'm not using HAOS, so I am just using the normal version. Is the addon version better supported or something?

Thats correct, `add-on` was designed with HAOS in mind, It has better support for validating parameters. `Standalone` and `Add-on` mode aren't that mutch different in the way they operate besides the parameters. 

In saying this this is the first time trying out the container via docker-hub and I seem to be running into some issues. Having a look at it now.

**Comment by werdnum** [comment]
As a meta question, what's the point in having a separate standalone type if the add-on version doesn't actually require running under HAOS? Would it make sense to maintain only the add-on version, or at least to strongly encourage preferring it over the standalone one?

**Comment by GeoDerp** [comment]
I think the best way to use add-on mode is via passing in the arguments via the standalone mode Image 
```
docker run -it -p 5000:5000 --name emhass-container -v $(pwd)/options.json:/app/options.json davidusb/emhass-docker-standalone --addon true --no_response True
```

**Comment by GeoDerp** [comment]
> As a meta question, what's the point in having a separate standalone type if the add-on version doesn't actually require running under HAOS? Would it make sense to maintain only the add-on version, or at least to strongly encourage preferring it over the standalone one?

Good question. The one main difference is that add-on mode calls HA via its local API to gain location and timezone information. 
In reality, it wouldn't be impossible to create an alternative method to check if its running in HAOS (to use the local API or not). Then standalone and add-on could be merged.

**Comment by davidusb-geek** [comment]
I will close this considering the latests improvements: battery and PV systems are optional.
We could probably still improve this by hiding the PV related paramters elsewhere when PV is turned off.
